usr/src/cmd/svc/milestone/fs-joyent

```diff
@@ -192,6 +192,14 @@ if [ $? -ne 0 ]; then
             mount -F lofs /usbkey/ssh /etc/ssh
         fi
 
+	#
+	# Set primarycache=metadata to avoid keeping swap data in RAM via the ARC.
+	# Set secondarycache=none to disable L2ARC for swap.
+	#
+	zfs set primarycache=metadata ${SYS_ZPOOL}/swap || \
+            fatal "failed to set primarycache=metadata on swap zvol"
+	zfs set secondarycache=none ${SYS_ZPOOL}/swap || \
+            fatal "failed to set secondarycache=none on swap zvol"
```

Comment:
Do you wish to do the same down below (line 220-ish) for dump?

Reply:
My gut says that dump is less critical than swap for the specific problem we're trying to solve, but I certainly could.
I was avoiding deciding how to document that bit without cluttering things.

Reply:
It can be simpler. Perhaps something like:

`# Dump volume only gets used during a kernel panic. Don't clutter it with any ARC.`

?

Reply:
As I'm thinking about it, I do think that IF we do this force-upgrade function, we should also change the dump device as well.

Reply:
I wish to reduce scope. Messing with the dump zvol is unlikely to help with the issue seen in production.
This simple change, even without touching the installers is likely to provide a lot of value for a minimal change.
I'm tying this to OS-8644 now instead of OS-8637

Reply:
Now that you have this (the fix-at-boot-time approach) as a distinct bug, a few more questions popped into my head:

- Do we want to be verbose/logging-verbose about "oh no, {primary,secondary}cache for swap set to `all`, setting to `{metadata,none}`?

- Do we want to have a way to override this?  Perhaps a `smartdc:swap-nofix` property in the `zones/swap` dataset?

Reply:
I'm trying desperately to think of any reason why anyone would ever want swap data blocks in the ARC.
This feels more to me like an oversight that we are finally fixing rather than adjusting a tunable from one reasonable setting to another. So my gut feeling is "no" to both. But I'm open to arguments why we might want to do one or the other or both.

Reply:
You've made your case for "no override" and I'm happy with it.

Still don't know if we want to be verbose on detection/fixing, though? It might be nice to see indications of "made change" in the SMF log of `filesystem/smartdc`.

Reply:
We didn't have those for the tweaks for the dump volume a little lower down which is what I modeled this after.
If you want them I will write them.

Reply:
Oh shoot, I missed the prior-art on this.  Okay, you've made your case for "no extra output" as well.